### ISSUE TITLE
custom loss function in keras

### ISSUE THREAD

**Issue by naserpiltan** [matterport/Mask_RCNN#1599]
Hi there.
i am writing my custom loss function for a detection purpose in keras.
my loss function returns a batch of tensors with size of 32.
each tensor contains 4 scalar number like this :
`[0.59131676 0.558732927 1.2102505 0.500861049]`
now i wanna feed my batches to model.compile as losses.
but i am a bit confused.
my question is that : does keras model.compile first calculate mean on my batch(i.e create a mean tensor) then sum all 4 numbers of mean tensor or first sum each tensor 4 scalars and then calculate mean of batch or there is an other operation??
i cant understand loss operation on a batch of tensors with size of > 1.
thank you.
